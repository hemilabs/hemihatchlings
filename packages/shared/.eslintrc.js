module.exports = {
  env: {
    es2021: true,
    node: true
  },
  extends: 'standard-with-typescript',
  overrides: [
  ],
  parserOptions: {
    ecmaVersion: 'latest',
    sourceType: 'module',
    project: ['tsconfig.json'],
    tsconfigRootDir: __dirname
  },
  plugins: [
    'sonarjs'
  ],
  rules: {
    '@typescript-eslint/triple-slash-reference': 'off',
    '@typescript-eslint/space-before-function-paren': 'off',
    '@typescript-eslint/strict-boolean-expressions': 'off',
    '@typescript-eslint/no-extraneous-class': 'off',
    'semi': ["error", "never"],
    'quotes': ["error", "single"],
    'import/extensions': 'off',
    'import/prefer-default-export': 'off',
    indent: ['error', 2],
    'class-methods-use-this': 'off',
    complexity: ['error', 4],
    'sonarjs/cognitive-complexity': ['error', 4],
    'max-depth': ['error', 3],
    'max-statements': ['error', 10],
    'max-lines': ['error', {
      max: 130,
      skipBlankLines: true,
      skipComments: true
    }],
    'max-lines-per-function': ['error', {
      max: 35,
      skipBlankLines: true,
      skipComments: true
    }],
    'max-nested-callbacks': ['error', 3],
    'max-params': ['error', 3],
    'no-useless-constructor': 'off',
    'max-classes-per-file': 'off',
    'lines-between-class-members': [
      'error',
      'always',
      {
        exceptAfterSingleLine: true
      }
    ],
    'no-shadow': 'off',
    '@typescript-eslint/no-shadow': ['error'],
    'max-len': ['error', {
      code: 80
    }],
    overrides: [
      {
        files: ['*.spec.ts'],
        rules: {
          'max-nested-callbacks': ['error', 5],
        }
      }
    ]
  }
}
