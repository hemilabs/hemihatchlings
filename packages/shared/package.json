{
  "name": "@hemihatchlings/shared",
  "version": "0.1.0",
  "description": "Shared package for hemihatchlings Application",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "type": "module",
  "private": true,
  "exports": {
    ".": "./src/index.ts"
  },
  "scripts": {
    "build": "tsc -p ./tsconfig.build.json",
    "lint": "eslint",
    "lint:fix": "eslint --fix",
    "test": "vitest run",
    "test:cov": "vitest run --coverage",
    "test:watch": "vitest watch"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "typescript": "^4.8.2",
    "uuid": "^9.0.0"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.19.0",
    "@types/node": "^18.7.15",
    "@types/uuid": "^8.3.4",
    "@typescript-eslint/eslint-plugin": "^5.0.0",
    "@vitest/coverage-c8": "^0.23.1",
    "eslint": "^8.0.1",
    "eslint-config-standard-with-typescript": "^22.0.0",
    "eslint-plugin-import": "^2.25.2",
    "eslint-plugin-n": "^15.0.0",
    "eslint-plugin-promise": "^6.0.0",
    "eslint-plugin-sonarjs": "^0.15.0",
    "ts-node": "^10.9.1",
    "vite": "^5.0.11",
    "vitest": "^0.23.1"
  }
}
