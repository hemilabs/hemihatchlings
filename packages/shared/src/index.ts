// Base Code
export * from "./domain/base/DomainError"
export * from "./domain/base/Entity"
export * from "./domain/base/ValueObject"

// Entities
export * from "./domain/entities/Hatchling"

// Enums
export * from "./domain/enums/ElementEnum"
export * from "./domain/enums/StageEnum"
export * from "./domain/enums/TransactionStatusEnum"

// Factories
export * from "./domain/factories/HatchlingFactory"

// Value Objects
export * from "./domain/valueObjects/Address"
export * from "./domain/valueObjects/Element"
export * from "./domain/valueObjects/Stage"
export * from "./domain/valueObjects/TransactionHash"
export * from "./domain/valueObjects/TransactionStatus"
export * from "./domain/valueObjects/Uuid"
