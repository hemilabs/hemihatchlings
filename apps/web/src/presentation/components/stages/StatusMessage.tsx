import { StageEnum } from '@hemihatchlings/shared'

interface StatusMessageProps {
  stage: StageEnum
  account: string
  transactionHash: string
}

function StatusMessage({
  stage,
  account,
  transactionHash
}:StatusMessageProps): JSX.Element {
  return (
    <>
      <div className="font-monaco text-sm text-white mt-2 text-center
        sm:text-base md:text-lg md:mt-4">
        <div>Wallet: {account}</div>
        <div>Transaction: {transactionHash}</div>
      </div>

      <div className={`${ stage != 'Adult' ? '' : 'hidden' }
        font-monaco text-sm text-white mt-2 text-center uppercase
        sm:text-base md:text-lg md:mt-4`}>
        Approaching { stage == StageEnum.Baby ? 'ETHEREUM' : 'BITCOIN' } Finality
      </div>
      <div className={`${ stage === 'Adult' ? '' : 'hidden' }
        font-monaco text-sm text-white mt-2 text-center uppercase
        sm:text-base md:text-lg md:mt-4`}>
        ethereum & bitcoin finality achieved
      </div>
    </>
  )
}

export default StatusMessage
