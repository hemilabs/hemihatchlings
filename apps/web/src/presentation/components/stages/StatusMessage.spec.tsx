import { describe, it, expect} from 'vitest'
import { RenderResult, render } from '@testing-library/react'
import StatusMessage from './StatusMessage'
import { StageEnum } from '@hemihatchlings/shared'

interface StatusMessageProps {
  stage: StageEnum
  account: string
  transactionHash: string
}

const renderComponent = ({
  stage, account, transactionHash
}:StatusMessageProps): RenderResult => {
    return render(
      <StatusMessage
        stage={stage}
        account={account}
        transactionHash={transactionHash} />
    )
  }

describe('src/presentation/StatusMessage', () => {
  const stage = StageEnum.Baby
  const account = '0x52908400098527886E0F7030069857D2E4169EE7'
  const transactionHash = '0x7304dc174aab2bc487b1befb9e35ba3632b9693f0c0548e138b4401f263910f1'

  it('renders correctly', () => {
    const { asFragment } = renderComponent({ stage, account, transactionHash })

    expect(asFragment()).toMatchSnapshot()
  })
})
