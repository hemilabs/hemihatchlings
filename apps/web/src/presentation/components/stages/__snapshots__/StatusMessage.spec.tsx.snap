// Vitest Snapshot v1

exports[`src/presentation/StatusMessage > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="
        font-monaco text-sm text-white mt-2 text-center uppercase
        sm:text-base md:text-lg md:mt-4"
  >
    Approaching ETHEREUM Finality
  </div>
  <div
    class="hidden
        font-monaco text-sm text-white mt-2 text-center uppercase
        sm:text-base md:text-lg md:mt-4"
  >
    ethereum & bitcoin finality achieved
  </div>
</DocumentFragment>
`;
